test/src/scheduler.js

```diff
@@ -319,4 +337,41 @@ describe('Scheduler', function() {
           assert.equal(tgThree.afterSecondReloadPageLoadCalled, true);
         });
   });
+
+  describe('saves assets when --save-assets is set', function() {
+    let prepareAssetsCalled = false;
+    let saveAssetsCalled = false;
+    const _prepareAssets = assets.prepareAssets;
+    assets.prepareAssets = function() {
+      prepareAssetsCalled = true;
+      return _prepareAssets.apply(this, arguments);
+    };
+    const _saveAssets = assets.saveAssets;
+    assets.saveAssets = function() {
+      saveAssetsCalled = true;
+      return _saveAssets.apply(this, arguments);
+    };
+
+    const screenshotGatherer = new TestScreenshotGatherer();
+    const gatherers = [screenshotGatherer];
+    const options = {
+      driver: fakeDriver,
+      url: 'https://example.com',
+      date: new Date(1464737670547),
+      flags: {
+        saveAssets: true
+      }
+    };
+
+    it('collects artifacts and goes through the scheduler', () => {
+      return scheduler.run(gatherers, options).then(_ => {
+        assert.equal(screenshotGatherer.postProfilingCalled, true);
+      });
+    });
+
+    it('asset saving functions are called from scheduler', () => {
+      assert.ok(prepareAssetsCalled);
+      assert.ok(saveAssetsCalled);
```

Comment:
yeah, testing that functionality may belong to the assetSaver test. My thinking was that you don't care how it saves, you just care that if you pass in the flag that it does save. Doing it this way tests that the assetSaver interface you've put together now is used as you're using it now, which isn't really what you care about, and so it might actually break in subtle ways in the future (e.g. a change makes the options not pass in correctly, so the functions are called but nothing is saved or something)

My preference is not include this test here but save it for a a component-level test (if we do those) of gathers + scheduler + saveAssets where it makes more sense, but your call